Try uncollapsing margins on the columns block.

jasmussen opened the pull request:
This is a first stab at fixing #7766.

It does not address it fully, I believe further work is necessary, including that of #7694. However this should at least improve the collapsing of the parent container.

In the following screenshots, I've selected all blocks, as this highlights using transparency and overlay effects, the _boundaries_ and how they overlap. Here's what's in `master`:

<img width="699" alt="before" src="https://user-images.githubusercontent.com/1204802/43386750-e95b5800-93e4-11e8-8a86-9abef8cc53a5.png">

Here's with this PR applied:

<img width="682" alt="after" src="https://user-images.githubusercontent.com/1204802/43386767-f44a3ed4-93e4-11e8-9760-6bde0050ae98.png">

As you can see, there are zeroed out margins on the first and last container children inside the columns block. This is there to make the columns block flow more directly with the surrounding content, but this can be revisited, and as noted, #7694 might help here.

Noting that if we were to _remove_ zeroed out margins on the first and last container children, here's what it would look like:

<img width="687" alt="no collapsing margins" src="https://user-images.githubusercontent.com/1204802/43386853-258e3fe0-93e5-11e8-835d-935b03cc0635.png">

Commit:
Try uncollapsing margins on the columns block.

This is a first stab at fixing #7766.

It does not address it fully, I believe further work is necessary, including that of #7694. However this should at least improve the collapsing of the parent container.

aduth reviewed (approved):
This seems like a small improvement over master, at least as far as overlapping goes. We should stop considering this as a "Columns" issue, and address it as generally affecting any block with inner blocks. In that sense, these changes do nothing for any other block with inner blocks.

aduth commented on core-blocks/columns/editor.scss:
```diff
@@ -32,12 +32,17 @@
 	margin-bottom: 0;
+
+	// This uncollapses margins on this parent container.
+	padding-top: 1px;
```
Would `0.1px` be a better value here? While it's not valid in the sense of a fraction of a pixel, it achieves the same effect while not having a visual difference (since it will round to `0`).

jasmussen commented:
> We should stop considering this as a "Columns" issue, and address it as generally affecting any block with inner blocks. In that sense, these changes do nothing for any other block with inner blocks.

Step one of making margins collapse was to ensure that the block "foot print" was accurate in the backend, compared to the front-end, specifically to ensure that the UI we paint around each block — borders, multi select squares, toolbars, side UI — did not affect the foot print at all.

Step two is actually adding margins to each and every block, **on a per-block basis**. Essentially, like a front-end styleseheet, a paragraph needs one margin, a list needs another. Perhaps a columns block doesn't need any margins at all. 

Right now we are applying the same margins in a blanket statement, to all blocks. This happens here: https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/block-list/style.scss#L140 

Removing that rule removes any artificial margins added to all the blocks (with the added notes that we have some negative margins to compensate for this). 

In other words, I would suggest that the next step in improving this, and addressing your comment, is to remove those artificial margins, and then rebuild them block by block. Do you agree this is the right approach?

jasmussen committed:
Move to .1px.
